### PR TITLE
Move Pact tests to GitHub Actions

### PR DESCRIPTION
**Pull request by ddzz**
## Description
This PR moves Pact tests from CircleCI to GitHub Actions. https://github.com/department-of-veterans-affairs/va.gov-team/issues/22374

## Testing done
- Pact tests run successfully on GitHub Actions

## Acceptance criteria
- [ ] Pact tests run successfully on GitHub Actions

### CODE REVIEW

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -91,6 +91,71 @@ jobs:
...
+        id: get-node-version
+        run: echo ::set-output name=NODE_VERSION::$(cat .nvmrc)
+
+      - name: Setup Node
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ steps.get-node-version.outputs.NODE_VERSION }}
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile --prefer-offline
```
Comment (billfienberg) [inline .github/workflows/continuous-integration.yml:124]: Do you have a link to any docs for `--prefer-offline`? This is the best I could find:
>`'--prefer-offline'`, 'use network only if dependencies are not available in local cache'

Source: https://github.com/yarnpkg/yarn/commit/d5a6f222e7ab7e25b9d8e1a12a158c24003ac677

**Review by billfienberg — APPROVED** [review]
Makes sense to me

File: .github/workflows/continuous-integration.yml (hunk 1 above)
Comment (U-DON) [reply .github/workflows/continuous-integration.yml:124]: It's not very well documented. I've only seen it as one of the CLI options from `yarn --help`.

File: .github/workflows/continuous-integration.yml [hunk 2]
```diff
@@ -91,6 +91,71 @@ jobs:
...
+        run: yarn pact:verify
+        env:
+          PACT_BROKER_BASIC_AUTH_USERNAME: pact-broker
+          PACT_BROKER_BASE_URL: https://dev.va.gov/_vfs/pact-broker/
+
+      - name: Run Pact publish
+        run: yarn pact:publish
+        env:
+          PACT_BROKER_BASIC_AUTH_USERNAME: pact-broker
+          PACT_BROKER_BASE_URL: https://dev.va.gov/_vfs/pact-broker/
```
Comment (U-DON) [inline .github/workflows/continuous-integration.yml:150]: Could potentially set `contract-tests.env` to use these variables throughout the job.

File: .github/workflows/continuous-integration.yml [hunk 3]
```diff
@@ -91,6 +91,71 @@ jobs:
...
+
+      - name: Archive contract test results
+        uses: actions/upload-artifact@v2
+        if: ${{ failure() }}
+        with:
+          name: contract-test-report
+          path: test-results/test-results.xml
```
Comment (U-DON) [inline .github/workflows/continuous-integration.yml:157]: I think it's worth saving `logs/pact.log` as well.

File: .github/workflows/continuous-integration.yml (hunk 2 above)
Comment (dginther) [reply .github/workflows/continuous-integration.yml:158]: You can also set env variables for the entire workflow

**Pull request merged**